### PR TITLE
Replace "User" role with "Partner" role

### PR DESCRIPTION
**Pull request by abbiefarr**
## Issue
#1616 
Also https://github.com/CityOfDetroit/bloom/issues/57

Addresses # (issue)

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

* Creates a new `user_roles` table, which holds user ids and their partner/admin permissions. 
* Adds users from the `listings_leasing_agents_user_accounts` to the new table as partners, as well as any admins from the `user_accounts` table.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Features manually tested:
* Partner login and application view
* Admin login
* Public listing view
* Resident login
* Resident application view

- [x] Desktop View
- [ ] Mobile View

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have updated the changelog to include a description of my changes

### CODE REVIEW

**Review by seanmalbert — COMMENTED** [review]
Hey @abbiefarr ,
I like your solution as it's a small departure from the current logic and meets the immediate need that the issue calls out. I'm not sure on how well this solution will scale to support more roles or roles with more attributes. At the same time, I don't think we have enough information to layout a proper proposal for a role manager that wouldn't potentially be overkill or create more technical debt, so I'm inclined to continue with this. Does Detroit have more complex needs further down the roadmap?

Please see my comment on the migration script, I think we can assign the partner role up front based off of data in the db.

File: backend/core/src/migration/1628276143069-partner-user-roles.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class partnerUserRoles1628276143069 implements MigrationInterface {
+    name = 'partnerUserRoles1628276143069'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "user_roles" ("is_admin" boolean NOT NULL DEFAULT false, "is_partner" boolean NOT NULL DEFAULT false, "user_id" uuid NOT NULL, CONSTRAINT "REL_87b8888186ca9769c960e92687" UNIQUE ("user_id"), CONSTRAINT "PK_87b8888186ca9769c960e926870" PRIMARY KEY ("user_id"))`);
+        await queryRunner.query(`ALTER TABLE "user_roles" ADD CONSTRAINT "FK_87b8888186ca9769c960e926870" FOREIGN KEY ("user_id") REFERENCES "user_accounts"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+        await queryRunner.query(`INSERT INTO "user_roles" ("user_id") SELECT "id" FROM "user_accounts"`)
+        // Initially, all existing users have partner access. This should be revoked over time.
```
[the remark this answers is not in the document]
Comment (abbiefarr) [reply backend/core/src/migration/1628276143069-partner-user-roles.ts]: Nice, thanks!

**Comment by abbiefarr** [comment]
Thanks for pointing me to the `listings_leasing_agents_user_accounts` table. Exactly what I was looking for!

Re: Detroit's future access control, I don't think they'll need anything more advanced than what would exist after this PR.

**Review by seanmalbert — APPROVED** [review]
Thanks, @abbiefarr , this looks good and the mirgration script works well! Can you resolve the merge conflict with the changelog, then I can get this in.

**Comment by seanmalbert** [comment]
Since I wanted to get this in for #1627, I branched this and resolved the conflict. So This got in on #1673 . @willrlin , not sure if this is possible, but if I can get permission to push to CityOfDetroit, I can handle merge conflicts on your team's PRs before merging.

**Comment by willrlin** [comment]
It looks like @abbiefarr did a merge from upstream yesterday in https://github.com/CityOfDetroit/bloom/pull/408. I assume that had everything that changed in #1673.

**Pull request closed without merging**